### ISSUE TITLE
Do not use reserved Python keywords as variable names

### ISSUE THREAD

**Issue by soupault** [scikit-image/scikit-image#4152]
## Description
This requires searching throught the complete codebase and renaming/deprecating old function/variable names.

The most commonly misused are `filter` and `slice`.

**Comment by rfezzani** [comment]
I wrote the PR #4158 to demonstrate how this issue can be managed using a decorator. Let me know if this kind of solution is acceptable or not ;-)

**Comment by rfezzani** [comment]
I also found `next` used in `skimage/morphology/greyreconstruct.py` :smile:

**Comment by rfezzani** [comment]
Do we consider that `array` is an acceptable argument name? I found an occurrence in `graph.mcp.route_through_array` for example.

**Comment by jni** [comment]
I would say yes, array is not reserved.

**Comment by soupault** [comment]
I agree with Juan. Although, it is probably not the best choice for the
variable name, since it is not descriptive.
